main.go

```diff
@@ -22,59 +29,178 @@ func main() {
-				if !ok {
-					return true
-				}
-				selExpr, ok := callExpr.Fun.(*ast.SelectorExpr)
-				if !ok {
-					return true
-				}
-				funcObj := pkg.TypesInfo.Uses[selExpr.Sel]
-				signature, ok := funcObj.Type().(*types.Signature)
-				if !ok {
-					return true
-				}
-				res, ok := signature.Results().At(0).Type().(*types.Named)
-				if !ok {
-					return true
-				}
-				ures, ok := res.Underlying().(*types.Struct)
-				if !ok {
-					return true
-				}
-				futureType, ok := ures.Field(0).Type().(*types.Named)
-				if !ok {
-					return true
+			if len(asnmt.Lhs) != 2 || len(asnmt.Rhs) != 1 {
+				return true
+			}
+
+			ret1, ok := asnmt.Lhs[0].(*ast.Ident)
+			if !ok {
+				return true
+			}
+
+			if ret1.Name != "_" {
+				return true
+			}
+
+			callExpr, ok := asnmt.Rhs[0].(*ast.CallExpr)
+			if !ok {
+				return true
+			}
+			selExpr, ok := callExpr.Fun.(*ast.SelectorExpr)
+			if !ok {
+				return true
+			}
+
+			if pkg.TypesInfo == nil {
+				return true
+			}
+			funcObj, ok := pkg.TypesInfo.Uses[selExpr.Sel]
+			if !ok {
+				return true
+			}
+
+			signature, ok := funcObj.Type().(*types.Signature)
+			if !ok {
+				return true
+			}
+			res, ok := signature.Results().At(0).Type().(*types.Named)
+			if !ok {
+				return true
+			}
+			ures, ok := res.Underlying().(*types.Struct)
+			if !ok {
+				return true
+			}
+			futureType, ok := ures.Field(0).Type().(*types.Named)
+			if !ok {
+				return true
+			}
+			future := futureType.Obj()
+			if future.Name() != "FutureAPI" || future.Pkg().Path() != "github.com/Azure/go-autorest/autorest/azure" {
+				return true
+			}
+
+			fmt.Println("Track 1 Hit: " + strings.TrimPrefix(pkg.Fset.Position(asnmt.Pos()).String(), rootPath+string(filepath.Separator)))
+
+			return false
+		})
+	}
+}
+
+func pandoraSDKScan(pkg *packages.Package, rootPath string) {
+	usesSet := map[types.Object][]string{}
+
+	// Collect Uses
+	for ident, def := range pkg.TypesInfo.Uses {
+		if !strings.Contains(ident.Name, "Create") &&
+			!strings.Contains(ident.Name, "CreateOrUpdate") &&
+			!strings.Contains(ident.Name, "Update") &&
+			!strings.Contains(ident.Name, "Delete") {
+			continue
+		}
+
+		// Only identify those Sync calls in Uses
+		if strings.Contains(ident.Name, "ThenPoll") {
+			continue
+		}
+
+		// Only identify Methods in Uses
+		_, ok := def.Type().(*types.Signature)
+		if !ok {
+			continue
+		}
+
+		usesSet[def] = append(usesSet[def], strings.TrimPrefix(pkg.Fset.Position(ident.Pos()).String(), rootPath+string(filepath.Separator)))
+	}
+
+	// Collect SDK
+	var sdkPkgPathList []string
+	for importedPkgPath := range pkg.Imports {
+		// Only target at Pandora SDK pkg import
+		if strings.Contains(importedPkgPath, "github.com/hashicorp/go-azure-sdk/resource-manager/") {
+			sdkPkgPathList = append(sdkPkgPathList, importedPkgPath)
+		}
+	}
+
+	if len(sdkPkgPathList) == 0 {
+		return
+	}
+
+	sdkSyncSet := map[types.Object]string{}
+	sdkAsyncSet := map[string]bool{}
```

Comment:
Instead, what we really need is a single set of type `map[types.Object]bool`, where the key is the sync function's object, and the value represents whether it has a async version.